docs/index.html

```diff
@@ -20,396 +20,490 @@
-        const [imageList, setImageList] = useState([]);
-        const [annotationsFolderHandle, setAnnotationsFolderHandle] = useState(null);
-        const [annotationsList, setAnnotationsList] = useState([]);
-        const [isLoadingWorker, setIsLoadingWorker] = useState(true);
-        const [isLoadingImages, setIsLoadingImages] = useState(false);
-        const [isLoadingAnnotations, setIsLoadingAnnotations] = useState(false);
-        const [annotationURL, setAnnotationURL] = useState("");
-        const [copyFeedback, setCopyFeedback] = useState("");
-        const [activeTab, setActiveTab] = useState("Local Deployment");
-
-        useEffect(() => {
-            const workerManager = new PyodideWorkerManager();
-            workerManager.createWorker().then(worker => {
-                setPyodideWorker(worker);
-                setIsLoadingWorker(false); // Hide spinner when worker is ready
-            }).catch(error => {
-                console.error("Error creating Pyodide worker:", error);
-            });
-
-            fetch(`./${PYTHON_FILE_NAME}`)
-                .then(response => response.text())
-                .then(text => {
-                    setCode(text);
-                })
-                .catch(error => {
-                    console.error(`Error loading ${PYTHON_FILE_NAME}:`, error);
-                });
-        }, []);
+    <script type="module">
+        import { get, set } from 'https://cdn.jsdelivr.net/npm/idb-keyval@6/+esm';
+        window.getFromDB = get;
+        window.saveToDB = set;
+    </script>
+    <div id="app"></div>
+    <script type="text/babel" data-presets="env,react">
 
-        const loginCallback = (context) => {
-            window.open(context.login_url);
-        }
+        const { useState, useEffect } = React;
+
+        const REGISTER_SERVICE_FILE = 'services.py';
+        let workerPromise;
+        function App() {
+            const [serverURL, setServerURL] = useState("wss://hypha.aicell.io/ws");
+            const [pyodideOutput, setPyodideOutput] = useState([]);
+            const [user, setUser] = useState({});
+            const [imageFolderHandle, setImageFolderHandle] = useState(null);
+            const [imageList, setImageList] = useState([]);
+            const [annotationsFolderHandle, setAnnotationsFolderHandle] = useState(null);
+            const [annotationsList, setAnnotationsList] = useState([]);
+            const [isRunning, setIsRunning] = useState(false);
+            const [isLoadingImages, setIsLoadingImages] = useState(false);
+            const [isLoadingAnnotations, setIsLoadingAnnotations] = useState(false);
+            const [annotationURL, setAnnotationURL] = useState("placeholder");
+            const [copyFeedback, setCopyFeedback] = useState("");
+            const [activeTab, setActiveTab] = useState("Local Deployment");
+
+            useEffect(() => {
+                workerPromise = new Promise((resolve, reject) => {
```

Comment:
I used a promise here, so we can defer the loading process, only wait for it when the user click deploy annotation. You can see how Promise works.